httpx/middleware/recovery.go

```diff
@@ -33,20 +33,15 @@ func (h *Recovery) ServeHTTPContext(ctx context.Context, w http.ResponseWriter,
 	ctx = reporter.WithReporter(ctx, h.Reporter)
 
 	// Add the request to the context.
-	reporter.AddRequest(ctx, r)
+	ctx = errctx.WithRequest(ctx, r)
 
 	// Add the request id
-	reporter.AddContext(ctx, "request_id", httpx.RequestID(ctx))
+	ctx = errctx.WithInfo(ctx, "request_id", httpx.RequestID(ctx))
 
 	defer func() {
-		if v := recover(); v != nil {
+		if e := errctx.Recover(ctx, recover()); e != nil {
```

Comment:
nit: why not use `err` here directly?

```
if err = errctx.Recover(ctx, recover()); err != nil {
  ...
}
```

Reply:
Because I don't want to override `err` if there was no panic, but the handler returned an error